libs/cli/langchain_cli/namespaces/app.py

```diff
@@ -55,21 +54,55 @@ def new(
             is_flag=True,
         ),
     ] = None,
+    nointeractive: Annotated[
+        bool,
+        typer.Option(
+            "--nointeractive",
+            help="Don't prompt for any input",
+            is_flag=True,
+        ),
+    ] = False,
 ):
     """
     Create a new LangServe application.
     """
     has_packages = package is not None and len(package) > 0
-    pip_bool = False
-    if pip is None and has_packages:
-        pip_bool = typer.confirm(
-            "Would you like to `pip install -e` the template(s)?",
-            default=False,
+
+    if nointeractive:
+        if name is None:
+            raise typer.BadParameter("name is required when --nointeractive is set")
+        name_str = name
+        pip_bool = bool(pip)
+    else:
+        name_str = (
+            name if name else typer.prompt("What folder would you like to create?")
         )
+        if not has_packages:
+            package = []
+            package_prompt = "What package would you like to add? (leave blank to skip)"
```

Comment:
We don't have a search interface here right? Would it be possible to add example syntax for packages? (I actually don't know the answer to this, but are the only supported packages the one's in templates directory or is there a notation for packages that are located elsewhere?)

Reply:
Not yet. And can add packages in either of these formats (from `langchain app add --help` which I realize I need to fix newlines in)
```
langchain app add extraction-openai-functions 
langchain app add \
 git+ssh://git@github.com/efriis/simple-pirate.git
```

Reply:
Added `langchain template list` which you can also search with e.g. `langchain template list rag`